pkg/plugins/runtime/gateway/gateway_route_generator.go

```diff
@@ -95,6 +95,10 @@ func GenerateEnvoyRouteEntries(host GatewayHost) []route.Entry {
 			// Make sure the prefix has a trailing '/' so that it only matches
 			// complete path components.
 			e.Match.PrefixPath = exactPath + "/"
+			if rw := e.Rewrite; rw != nil && rw.ReplacePrefixMatch != nil {
+				prefix := strings.TrimRight(*rw.ReplacePrefixMatch, "/") + "/"
+				rw.ReplacePrefixMatch = &prefix
```

Comment:
Why did we call that `ReplacePrefixMatch`? Envoy calls it `PrefixRewrite`, it seems that it's just an unnecessary mapping I have to do in my head.

Reply:
[Gateway API](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io%2fv1beta1.HTTPPathModifier)

We aren't _just_ an Envoy wrapper :shrug:

Reply:
Oh, if it's in Gateway API then 👍 - I just thought that we had a different name for the sake of having a different name. Thanks for clarifying! 

> just an Envoy wrapper

I know but if there is a 1:1 relation with some fields I think it's ok to re-use the name to make it easier to read, WDYT?

Reply:
> I know but if there is a 1:1 relation with some fields I think it's ok to re-use the name to make it easier to read, WDYT?

Agreed in general but this is the case I think where the relation isn't 1-1, here we're still in the domain of Gateway API/MeshGatewayRoute matchers and transforming MeshGatewayRoute matchers into other MeshGatewayRoute matchers, the transformation to Envoy config happens later.